metric_namer_test.go

```diff
@@ -21,259 +21,982 @@ package otlptranslator
+		},
+		{
+			name: "metric with per hour unit",
+			namer: MetricNamer{
+				UTF8Allowed:        false,
+				WithMetricSuffixes: true,
+			},
+			metric: Metric{
+				Name: "events",
+				Unit: "1/h",
+				Type: MetricTypeGauge,
+			},
+			expected: "events_per_hour",
+		},
+		{
+			name: "metric with per day unit",
+			namer: MetricNamer{
+				UTF8Allowed:        false,
+				WithMetricSuffixes: true,
+			},
+			metric: Metric{
+				Name: "transactions",
+				Unit: "1/d",
+				Type: MetricTypeGauge,
+			},
+			expected: "transactions_per_day",
+		},
+		{
+			name: "metric with per week unit",
+			namer: MetricNamer{
+				UTF8Allowed:        false,
+				WithMetricSuffixes: true,
+			},
+			metric: Metric{
+				Name: "reports",
+				Unit: "1/w",
+				Type: MetricTypeGauge,
+			},
+			expected: "reports_per_week",
+		},
+		{
+			name: "metric with per month unit",
+			namer: MetricNamer{
+				UTF8Allowed:        false,
+				WithMetricSuffixes: true,
+			},
+			metric: Metric{
+				Name: "invoices",
+				Unit: "1/mo",
+				Type: MetricTypeGauge,
+			},
+			expected: "invoices_per_month",
+		},
+		{
+			name: "metric with per year unit",
+			namer: MetricNamer{
+				UTF8Allowed:        false,
+				WithMetricSuffixes: true,
+			},
+			metric: Metric{
+				Name: "renewals",
+				Unit: "1/y",
+				Type: MetricTypeGauge,
+			},
+			expected: "renewals_per_year",
+		},
+		{
+			name: "metric with unknown per unit",
+			namer: MetricNamer{
+				UTF8Allowed:        false,
+				WithMetricSuffixes: true,
+			},
+			metric: Metric{
+				Name: "custom",
+				Unit: "1/custom_unit",
+				Type: MetricTypeGauge,
+			},
+			expected: "custom_per_custom_unit",
+		},
+
+		// Edge cases with empty and whitespace units
+		{
+			name: "metric with empty per unit",
+			namer: MetricNamer{
+				UTF8Allowed:        false,
+				WithMetricSuffixes: true,
+			},
+			metric: Metric{
+				Name: "metric",
+				Unit: "By/",
+				Type: MetricTypeGauge,
+			},
+			expected: "metric_bytes",
+		},
+		{
+			name: "metric with whitespace in unit",
+			namer: MetricNamer{
+				UTF8Allowed:        false,
+				WithMetricSuffixes: true,
+			},
+			metric: Metric{
+				Name: "metric",
+				Unit: " By / s ",
+				Type: MetricTypeGauge,
+			},
+			expected: "metric_bytes_per_second",
+		},
+		{
+			name: "metric with only slash in unit",
+			namer: MetricNamer{
+				UTF8Allowed:        false,
+				WithMetricSuffixes: true,
+			},
+			metric: Metric{
+				Name: "metric",
+				Unit: "/",
+				Type: MetricTypeGauge,
+			},
+			expected: "metric",
+		},
 	}
 
-	for _, test := range tests {
-		mainUnitSuffix, perUnitSuffix := buildUnitSuffixes(test.unit)
-		require.Equal(t, test.expectedMain, mainUnitSuffix)
-		require.Equal(t, test.expectedPer, perUnitSuffix)
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := tt.namer.Build(tt.metric)
+			if got != tt.expected {
+				t.Errorf("MetricNamer.Build() = %q, want %q", got, tt.expected)
+			}
```

Comment:
@aknuds1, this is the single piece of code in the whole file that compares `got` and `expected`. Sure, I can use `require` here and create a new PR replacing `require` with this if comparison, but that's such a small change... is the PR really unreviewable because of these three lines?